### ISSUE TITLE
Binary Distribution

### ISSUE THREAD

**Issue by MisterY** [hpdeifel/hledger-iadd#41]
It would be great if iadd was distributed as a binary package for those who have to install all the development tools just for compiling one addon.
I'm in need of one for Windows and Linux. On Linux, it should be possible to package an AppImage or another distribution-independent binary package.

Unfortunately, I can't currently try it out but in the screenshots it looks great. :)

**Comment by hpdeifel** [comment]
I agree! There already is a binary for Archlinux on the release page, you could try if it works on your system. Unfortunately I don't have Windows but I can look into distro-independent linux packaging.

**Comment by simonmichael** [comment]
Homebrew is a not so bad option these days for mac/linux/windows. There's a hledger formula you could add to.

Nix too, but not for windows.

**Comment by the-solipsist** [comment]
@simonmichael, when you're building the Ubuntu packages for `hledger`, `hledger-ui`, and `hledger-web`, could you also possibly build a package for `hledger-iadd`? Or would that be complicated?

**Comment by simonmichael** [comment]
Yes, I could do that in principle if @hpdeifel wants it. I feel like there would be some complications with bounds at times. Usually these binary artifacts are built when I push to ci-linux/ci-mac/ci-windows branches on github. That mostly happens a little before release time, but not only then.

**Comment by simonmichael** [comment]
I suppose it could make a best effort and tolerate build failures, to avoid breaking the CI.

**Comment by hpdeifel** [comment]
Sure, if it's not too much overhead for you, that would be great!

**Comment by the-solipsist** [comment]
Update: the binary in the `.tar.xz` archive released by @hpdeifel is working fine on an x86_64 Ubuntu 20.04 machine.  (I've just inflated the binary to `~/.local/bin/hledger-iadd`).  It would be great if @simonmichael could add that binary to the `hledger-ubuntu.zip` file along with `hledger-ui` and `hledger-web` as part of the hledger [release workflow](https://github.com/simonmichael/hledger/actions/).  Thanks!
